code/controllers/subsystems/payment.dm

```diff
@@ -85,9 +94,13 @@ SUBSYSTEM_DEF(payment_controller)
 	for(var/mob/living/carbon/H in GLOB.living_players) //we don't pay dead people
 		if(!H.mind?.assigned_role || !H.mind?.initial_account) //this excludes stowaways and other non-crew roles including derelict ships
 			continue
-
 		if(H.mind.assigned_role == "Captain" && GLOB.using_map.name == "Nerva") //todo, make a map var for captain ownership if we have other ship maps in a similar situation (never)
 			continue
+		if(H.mind.assigned_role == "Passenger")
+			passenger_count += 1
+			var/datum/transaction/U = new(H.mind.initial_account, station_account, passenger_fee, "[GLOB.using_map.station_name] Automated Fee System")
```

Comment:
I _generally_ don't like using an actual transaction here, as opposed to a singular.
This is so that on the station account, you can group all fees under a single transaction, otherwise it'll be spammed with each individual one.

If you wanted to go the singular route, you could do the passenger side here as normal, then the station account side at LN80 or so, and just do `passenger_fee * paying_passengers`

Reply:
I don't think there will ever be enough passengers or transactions to result in true spam... also every attempt I've made of this today resulted either in runtimes, bad transactions, or bad transaction origin names (/datum/station_account instead of "Automated..."

Reply:
That's because it's expecting a string, not an account datum. You can look at what the payroll does as an example.

Reply:
Getting the origin to display properly throws the runtimes as it starts trying to check .suspended on every payday and failing

Reply:
That's because the `singluar` subtype is very weird. The first arguement is _always_ the real account, regardless of the direction the funds are moving in. The direction (debit or credit) is effected by the amount being positive or not.